### DISCUSSION TITLE
all_details not return reasons

### DISCUSSION THREAD

**Discussion by bragovo** [palkan/action_policy#181] [Q&A]
Hi there!

In my policy I used `deny!(:login)` and in exception I can't find reason in `all_details`:
```
exception.result.all_details
# {}

exception.result.reasons.details.values.flatten
# [:login]
```

action_policy 0.5.7
What I did wrong?

**Comment by palkan** [comment]
There is some inconsistency (or a naming issue): `reasons.details` is an equivalent of ActiveModel `error.details`, which contains all the reasons as a Hash.

On the other hand, `all_details` contains only the [**detailed reasons**](https://actionpolicy.evilmartians.io/#/reasons?id=detailed-reasons)., i.e., those with additional information.

When you call `deny!`, you add a _regular reason_, not a detailed one.

We can say that `all_details` and `details` in a policy class are to deal with extra metadata. While `reasons.details` is just a serialized representation of the reasons. We should rename `reasons.details` to something else, I think.
